torch/__init__.py

```diff
@@ -1308,20 +1308,60 @@ def compiled_with_cxx11_abi():
     lstsq,
 )
 
+
 class _TorchCompileInductorWrapper:
     compiler_name = "inductor"
 
-    def __init__(self, mode, passes):
-        from torch._dynamo.eval_frame import lookup_backend
-        from torch._inductor.config import InductorConfigContext
-
-        self.compile_fn = lookup_backend(self.compiler_name)
-        self.cm = InductorConfigContext(mode if mode is not None else passes)
-        self._torchdynamo_orig_callable = self.compile_fn
+    def __init__(self, mode, passes, dynamic):
+        from torch._inductor.compile_fx import compile_fx
+
+        self.compile_fn = compile_fx
+        self._torchdynamo_orig_callable = compile_fx
+        self.config = dict()
+        self.apply_mode(mode)
+        self.apply_passes(passes)
+        if dynamic:
+            # cudagraphs conflicts with dynamic shapes
+            self.config["triton.cudagraphs"] = False
```

Comment:
Instead of silently contradicting the config, it feels like we should assert and force users to set it False